x-pack/solutions/observability/plugins/observability_agent/server/utils/get_observability_data_sources.ts

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { CoreSetup, Logger } from '@kbn/core/server';
+import type { APMIndices } from '@kbn/apm-sources-access-plugin/server';
+import { getLogsIndices } from './get_logs_indices';
+import { getApmIndices } from './get_apm_indices';
+import type {
+  ObservabilityAgentPluginSetupDependencies,
+  ObservabilityAgentPluginStart,
+  ObservabilityAgentPluginStartDependencies,
+} from '../types';
+
+export async function getObservabilityDataSources({
+  core,
+  plugins,
+  logger,
+}: {
+  core: CoreSetup<ObservabilityAgentPluginStartDependencies, ObservabilityAgentPluginStart>;
+  plugins: ObservabilityAgentPluginSetupDependencies;
+  logger: Logger;
+}): Promise<{
+  apmIndexPatterns: APMIndices;
+  logIndexPatterns: string[];
+  metricIndexPatterns: string[];
+  alertsIndexPattern: string[];
+}> {
+  const apmIndexPatterns = await getApmIndices({ core, plugins, logger });
+  const logIndexPatterns = await getLogsIndices({ core, logger });
+  const metricIndexPatterns = ['metrics-*'];
```

Comment:
We should add the metrics equivalent in a followup.

Reply:
I see now that `x-pack/solutions/observability/plugins/metrics_data_access` exists We should use that similar to `apmDataAccess` and `logsDataAccess`